firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -105,11 +105,12 @@ public void listenForNotifications() {
         handleNotifications(inputStream);
         inputStream.close();
       } catch (IOException ex) {
-        propagateErrors(
-            new FirebaseRemoteConfigClientException(
-                "Unable to parse config update message.",
-                ex.getCause(),
-                FirebaseRemoteConfigException.Code.CONFIG_UPDATE_MESSAGE_INVALID));
+        // Stream was interrupted due to a transient issue and the system will retry the connection.
+        Log.d(
+            TAG,
+            String.format(
+                "Stream was cancelled due to an exception: %s. Retrying the connection...",
```

Comment:
Sorry last nit 😅  Was thinking about the exception in the middle of the log and remembered `Log.d` can take an exception as a parameter and print it, so we can use those for the logs here: https://developer.android.com/reference/android/util/Log.html#d(java.lang.String,%20java.lang.String,%20java.lang.Throwable)

Reply:
Added it to the logs